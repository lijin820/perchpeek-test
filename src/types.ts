export type SelectType = {
  value: string;
  label: string;
  key: string;
};

export type ModalType = {
  visible: boolean;
  setVisble: (args: boolean) => void;
};
