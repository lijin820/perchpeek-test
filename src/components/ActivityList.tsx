import React, { ReactElement } from "react";
import { List, Typography } from "antd";
import { ListType } from "../types";

function ActivityList({ activities }: ListType): ReactElement {
  return (
    <List
      data-id="list"
      style={{ marginTop: 20 }}
      bordered
      dataSource={activities}
      renderItem={(item) => (
        <List.Item data-id="list-item">
          <Typography.Text strong>
            You&apos;ve created a {item.type}:
          </Typography.Text>
          {item.url} {item.note}
        </List.Item>
      )}
    />
  );
}

export default ActivityList;
