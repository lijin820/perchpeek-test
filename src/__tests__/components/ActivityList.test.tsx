import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import ActivityList from "../../components/ActivityList";
import { ActivityType } from "../../types";

describe("ActivityList", () => {
  const initialState = { activities: [] };
  const mockStore = configureStore();
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let store: any;
  const activities: ActivityType[] = [];

  describe("rendering", () => {
    store = mockStore(initialState);
    let wrapper: ShallowWrapper;
    beforeEach(() => {
      wrapper = shallow(
        <Provider store={store}>
          <ActivityList activities={activities} />
        </Provider>
      );
    });

    it("renders the component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("renders parent component", () => {
      expect(wrapper.find(ActivityList)).toHaveLength(1);
    });
  });
});
