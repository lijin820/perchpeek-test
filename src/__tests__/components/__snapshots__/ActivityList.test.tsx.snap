// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityList rendering renders the component 1`] = `
<List
  bordered={true}
  data-id="list"
  dataSource={
    Array [
      Object {
        "note": "test note",
        "type": "test type",
        "url": "test url",
      },
    ]
  }
  renderItem={[Function]}
  style={
    Object {
      "height": 300,
      "marginTop": 20,
      "overflowY": "auto",
    }
  }
/>
`;
